HGNC_Genomic_Files_Migration/hgnc_data_file_migration.py

```diff
@@ -0,0 +1,243 @@
+	output_mode = parser.add_mutually_exclusive_group()
+	output_mode.add_argument('-l', '--stdout_log', required=False, action = 'store_true', help='Dry-run the script. Preview the changes that will be made to the files.')
+	output_mode.add_argument('-o', '--override_file', required=False, action = 'store_true', help='Override the old data files.')
+	output_mode.add_argument('-n', '--create_new_file', required=False, action = 'store_true', help='Save the migrated data to new file without overriding the old files.')
+	args = parser.parse_args()
+	return args
+
+#---Read hgnc gene, hgnc alias tables and outdated to new entrez id mappings--->
+#---Currently reading from a file. This logic will be replaced to read the gene/alias data from cbioportal API once the new hgnc gene tables become available--->
+def hgnc_gene_data():
+	print("\nFetching the HGNC reference gene and alias info..\n")
+	main_table_df = pd.read_csv('hgnc_main_table.txt', comment='#', sep='\t', header=0, keep_default_na=False, dtype=str, low_memory=False)
+	alias_table_df = pd.read_csv('hgnc_alias_table.txt', comment='#', sep='\t', header=0, keep_default_na=False, dtype=str, low_memory=False)
+	alias_table_df1 = alias_table_df.groupby(['entrez_id'],sort=False)['symbol'].unique().apply(list).reset_index()
+
+	main_table_entrez_dict = dict(zip(main_table_df['entrez_id'],main_table_df['symbol']))
+	alias_table_entrez_dict = dict(zip(alias_table_df1['entrez_id'],alias_table_df1['symbol']))
+
+	#Read the outdated to new entrez id mapping file (consolidated from all sub categories of the analysis)
+	outdated_entrez_df = pd.read_csv('outdated_entrez_ids.txt', sep='\t', header=0, keep_default_na=False, dtype=str)
+	outdated_entrez_dict = dict(zip(outdated_entrez_df['old_entrez_id'],outdated_entrez_df['updated_entrez_id']))
+	
+	#Read the outdated to new hugo symbol mapping file (for cases where the file has only hugo symbol column)
+	outdated_hugo_df = pd.read_csv('outdated_hugo_symbols.txt', sep='\t', header=0, dtype=str)
+	outdated_hugo_dict = dict(zip(outdated_hugo_df['outdated_hugo_symbol'],outdated_hugo_df['new_hugo_symbol']))
+	
+	return main_table_entrez_dict, alias_table_entrez_dict, outdated_entrez_dict, outdated_hugo_dict
+
+#---Check the input type (file/directory) and get the list of files (1 or many)--->
+def check_path(source_path):
+	files_list = []
+	exluded_files_list = ['data_bcr_clinical_data_patient.txt','data_bcr_clinical_data_sample.txt','data_clinical_patient.txt','data_clinical_sample.txt','data_clinical_supp_hypoxia.txt','data_gene_matrix.txt','data_microbiome.txt','data_mutational_signature_confidence.txt','data_mutational_signature_contribution.txt','data_timeline_labtest.txt','data_timeline_procedure.txt','data_timeline_specimen.txt','data_timeline_status.txt','data_timeline_surgery.txt','data_timeline_treatment.txt','data_timeline.txt','data_subtypes.txt','data_fusions.txt']
```

Comment:
`data_fusion` files should be updated as well for outdated symbols?

Reply:
Updated the script to handle `data_fusions`. The symbols in `Fusion` column gets updated as well along with `Hugo_Symbol` column.